File: app/src/main/java/jp/ttanaka330/androidlearning/service/WeatherApi.java
package jp.ttanaka330.androidlearning.service;

import io.reactivex.Observable;
import jp.ttanaka330.androidlearning.service.entity.WeatherResponse;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @see <a href="https://openweathermap.org/">OpenWeatherMap</a>
 */
public interface WeatherApi {

    String ENDPOINT = "http://api.openweathermap.org";
    String ICON_URL = "http://openweathermap.org/img/w/";
    String API_KEY = "6505d0e6056d1807d2a13859bf1caf2e";

    @GET("/data/2.5/forecast")
    Observable<WeatherResponse> getWeather(@Query("appid") String appId, @Query("id") String cityId);

}

File: learning/webapi/openweathermap/src/main/java/com/github/ttanaka330/learning/webapi/openweathermap/data/WeatherApi.kt
package com.github.ttanaka330.learning.webapi.openweathermap.data

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @see <a href="https://openweathermap.org/">OpenWeatherMap</a>
 */
interface WeatherApi {

    companion object {
        const val ENDPOINT = "http://api.openweathermap.org"
        const val ICON_URL = "http://openweathermap.org/img/w/"
        const val API_KEY = "6505d0e6056d1807d2a13859bf1caf2e"
    }

    @GET("/data/2.5/forecast")
    fun getWeather(
        @Query("appid") appId: String,
        @Query("id") cityId: String
    ): Observable<WeatherForecastResponse>
}